AK/URLParser.cpp

```diff
@@ -16,6 +16,10 @@
 
 namespace AK {
 
+// NOTE: This just makes EOF unsigned instead of -1.
+#undef EOF
+#define EOF 0xFFFFFFFF
```

Comment:
This seems like a completely unnecessary reuse of the common `EOF` name.
Can we just call it `end_of_file` or something instead?
Also please use a `constexpr u32` instead of a macro :)